clever-tools/default.nix

```diff
@@ -1,36 +1,58 @@
-{ stdenv, fetchurl, glibc, nodegit }:
+{ stdenv, fetchurl, glibc }:
 
 stdenv.mkDerivation rec {
   name = "clever-tools-${version}";
-  version = "0.9.3";
+  version = "1.1.0";
 
   src = fetchurl {
     url = "https://clever-tools.cellar.services.clever-cloud.com/releases/${version}/clever-tools-${version}_linux.tar.gz";
-    sha256 = "adcae5af912dcbdc74d996b6e94767f24d16bf1bdcd5073797f999fe75b018a4";
+    sha256 = "1lkwckmlz2gf41xmnzydpcbly0jcry3bn8lp570gha768ic7xmb1";
   };
 
-  buildInputs = [ nodegit ];
-
   libPath = stdenv.lib.makeLibraryPath [ stdenv.cc.cc glibc ];
 
-  nodegitLibrary = stdenv.lib.makeLibraryPath [ nodegit ];
-
   installPhase = ''
-    tar --extract --file=$src linux/clever --transform 's/linux\///'
+    tar --extract --file=$src
     bin=$out/bin/clever
     mkdir -p $out/bin
     mv clever $bin
-    ln -s "$nodegitLibrary/nodegit.node" "$out/bin/nodegit.node"
   '';
 
   dontStrip = "true";
 
+  # NodeJS applications delivered with zeit/pkg contains hardcoded offsets LOOL
+  # see https://github.com/brendan-hall/nixpkgs/blob/e3b313bb59f49f10970205aafd44878d35da07e7/pkgs/development/web/now-cli/default.nix#L25
   preFixup = ''
     bin=$out/bin/clever
-    patchelf \
-      --set-rpath "$libPath" \
-      --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
-      "$bin"
+    orig_size=$(stat --printf=%s $bin)
+    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $bin
+    patchelf --set-rpath ${libPath} $bin
+    chmod +x $bin
+    new_size=$(stat --printf=%s $bin)
+    ###### zeit-pkg fixing starts here.
+    # we're replacing plaintext js code that looks like
```

Comment:
Do you understand what's going on there and why it's needed?

Reply:
The zeit/pkg is using hardcoded offset to call some entrypoint or w/e (I have no clue of why he's doing that, probably because it bundle some plain javascript into the executable). 
Because of patchelf being used to add `rpath` and fix the `interpreter`, we get a shifted binary (a little bigger than the original). And that interfere with the aforementioned `hardcoded offset` used by zeit/pkg.

Reply:
Ok, thanks